mail_filter_adressee_by_contact/models/mail_compose_message.py

```diff
@@ -23,30 +17,50 @@ class MailComposer(models.TransientModel):
         "Additional Contacts",
     )
 
-    apply_filter = fields.Boolean(default=True, string="Filtering relevant adressees")
+    apply_filter = fields.Selection(
+        (
+            [
+                ("contacts", "Contacts"),
+                ("users", "Users"),
+                ("all", "All"),
+            ]
+        ),
+        string="Filtering relevant adressees",
+        default="contacts",
+        required=True,
+    )
 
     @api.onchange("apply_filter")
     def _get_partner_ids_domain(self):
         domain = [("type", "!=", "private")]
         model = self._context.get("active_model")
-        if model in FILTERED_MODEL and self.apply_filter:
-            domain.insert(0, "&")
-            records = self.env[self._context["active_model"]].browse(
-                self._context.get("active_ids")
+        if self.apply_filter == "contacts":
+            method_name = f"_get_domain_for_{model.replace('.', '_')}"
```

Comment:
Model may not exist:

Steps to reproduce in runboat:
Go to Invoicing > Invoices > Select one > Go to chatter > Log note > Expand to open composer:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/odoo/http.py", line 685, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/odoo/http.py", line 361, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo/http.py", line 349, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo/http.py", line 914, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo/http.py", line 533, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/addons/web/controllers/main.py", line 1394, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/main.py", line 1386, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 399, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 386, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/odoo/models.py", line 6264, in onchange
    record._onchange_eval(name, field_onchange[name], result)
  File "/opt/odoo/odoo/models.py", line 6006, in _onchange_eval
    method_res = method(self)
  File "/mnt/data/odoo-addons-dir/mail_filter_adressee_by_contact/models/mail_compose_message.py", line 38, in _get_partner_ids_domain
    method_name = f"_get_domain_for_{model.replace('.', '_')}"
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 641, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/http.py", line 317, in _handle_exception
    raise exception.with_traceback(None) from new_cause
AttributeError: 'NoneType' object has no attribute 'replace'
```

Reply:
good point, I put back the module's list "FILTERED_MODEL" to avoid this.